### PR TITLE
Remove TPR code

### PR DESCRIPTION
**Pull request by rossf7**
Removes TPR code in preparation for updating deps to master.

### CODE REVIEW

File: service/alerter/service.go [hunk 1]
```diff
@@ -1,19 +1,17 @@
 package alerter
 
 import (
-	"encoding/json"
 	"fmt"
 	"strings"
 	"time"
 
-	"github.com/giantswarm/awstpr"
+	"github.com/giantswarm/apiextensions/pkg/clientset/versioned"
```
Comment (rossf7) [inline service/alerter/service.go]: Mainly just deleting old crap in this PR. Only changes were for the alerter service to use `apiextensions`.

File: service/resource/legacyv1/domains.go [hunk 2]
```diff
@@ -1,58 +0,0 @@
-package legacyv1
```
Comment (xh3b4sd) [inline service/resource/legacyv1/domains.go]: Why is the legacy resource vanishing now? This should actually stay until all existing guest clusters are not using it anymore. I am not aware that all AWS CROs have a version bundle now. Was there a migration?

File: service/cloudconfigv1/cloud_config.go [hunk 3]
```diff
@@ -1,105 +0,0 @@
-package cloudconfigv1
```
Comment (xh3b4sd) [inline service/cloudconfigv1/cloud_config.go]: Would be cool to migrate and clean up code in different PRs.

File: service/service.go [hunk 4]
```diff
@@ -58,11 +59,10 @@ func DefaultConfig() Config {
 
 type Service struct {
 	// Dependencies.
-	Alerter               *alerter.Service
-	CRDFramework          *framework.Framework
-	CustomObjectFramework *framework.Framework
```
Comment (xh3b4sd) [inline service/service.go]: In the boot function there is commented code which can be removed.

File: service/resource_router.go [hunk 5]
```diff
@@ -34,36 +33,3 @@ func NewResourceRouter(resources map[string][]framework.Resource) func(ctx conte
 		return enabledResources, nil
 	}
 }
-
-// NewTPRResourceRouter determines which resources are enabled based upon the
-// version in the version bundle. It will be removed when we remove TPR support.
-func NewTPRResourceRouter(resources map[string][]framework.Resource) func(ctx context.Context, obj interface{}) ([]framework.Resource, error) {
```
Comment (xh3b4sd) [inline service/resource_router.go]: I do not see how the resources are now managed. There is `NewResourceRouter` but it is nowhere used, or I cannot find it in the diff. I also do not see how the resources map is created.

File: service/resource/legacyv1/domains.go (hunk 2 above)
Comment (rossf7) [reply service/resource/legacyv1/domains.go]: This only removes `legacyv1` which uses TPRs. The `legacyv2` resource uses CRDs and is what is in use. The resource router defaults to `legacyv2` if the version bundle is not set so no migration is required.

**Comment by rossf7** [comment]
@xh3b4sd Ack. The PR is big. I'll split it and delete resource by resource so its clearer what is going.

File: service/resource/legacyv1/domains.go (hunk 2 above)
Comment (xh3b4sd) [reply service/resource/legacyv1/domains.go]: I see. That makes sense. Thanks.

**Comment by rossf7** [comment]
Closing. Done in #640

**Pull request closed without merging**